core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -0,0 +1,112 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package kafka.server
+
+import java.util.Properties
+
+import kafka.admin.AdminUtils
+import kafka.log.LogConfig
+import kafka.metrics.KafkaMetricsGroup
+import kafka.utils._
+import org.apache.kafka.common.errors.InvalidRequestException
+import org.apache.kafka.common.metrics.Metrics
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.requests.CreateTopicsRequest._
+
+import scala.collection._
+import scala.collection.JavaConverters._
+
+class AdminManager(val config: KafkaConfig,
+                   val metrics: Metrics,
+                   val metadataCache: MetadataCache,
+                   val zkUtils: ZkUtils) extends Logging with KafkaMetricsGroup {
+  this.logIdent = "[Admin Manager on Broker " + config.brokerId + "]: "
+
+  val topicPurgatory = DelayedOperationPurgatory[DelayedOperation]("topic", config.brokerId)
```

Comment:
"topic" seems to be an ambiguous name for this purgatory. If we expect this purgatory to be specific to topic creation, this probably should be createTopic. If we expect this to be generic to all admin tasks, this probably should be admin.